### DISCUSSION TITLE
DESrv 2.0 plans

### DISCUSSION THREAD

**Discussion by EgorBron** [Blusutils/DESrv#29] [Announcements]
## Hello, everyone!

DESrv has received more attention lately. Some unexpected ideas have been added.

However, DESrv didn't have a really correct and stable code base (at least because I was learning C# during DESrv API implementation). After half a year it feels hard to integrate new things. So soon **DESrv will be rewritten from scratch and released as version 2.0**.

> And I hope it will happen before May 27...

## About the changes
1. First of all, the loader and extension API (known as PDK) will be rewritten. At the moment the whole system is extremely unstable (almost every minor release breaks extensions even without any visible functionality). The loading of extensions itself in the Core part is also extremely inconvenient.
2. In 6 months, it has become clear that DESCEndLib is ballast. We almost don't need it in practice (logging, thread management, localization and configuration support are not needed in every project, and it would be better to use several third party libraries rather than just one). It's worth moving it back to the Core solution.
3. It was not originally planned to give extensions full access to work with connections, configs, translations, scripts, etc. However, in 2.0, we want to pay special attention to these moments.
4. Also not so long ago support for Linux and standalone builds appeared. In 2.0 we will add support for macOS and ~~good~~ working standalone/runtime-depended autobuild for all systems using Github Actions. Also, DESrv will work on latest .NET release.
5. Extensions may use third party libraries, but this has not been tested at this time. In the future there will be 100% support for third party libraries.
6. Support for native (C/C++) extensions. A crazy task, which was set and then dropped. However, it makes some sense and can be implemented in version 2.0.
7. Custom random number generators are useless and will be replaced by third party libraries.

**These are not all plans. In the first place, of course, is to write the project from scratch for .NET 7.0 and full support for connectivity and third-party libraries.**

## Development progress
You can watch the development progress in #30 (`desrv2.0` branch)

**Comment by stngularity** [comment]
Я так понимаю, говоря "Раз поток, два поток, есть говно в многопоток" ты имел введу эту версию?

**Reply by EgorBron** [reply]
> Я так понимаю, говоря "Раз поток, два поток, есть говно в многопоток" ты имел введу эту версию?
> *translated*
> I think when you say "First thread, second thread, we have shit in multithread" you mean this version?

No. I meant current version based on old CEnd. In 2.0 threads' logic will be rewritten completely.
